pf/tfgen/main.go

```diff
@@ -76,29 +73,17 @@ func Main(provider string, info tfbridge.ProviderInfo) {
 // This is an experimental API.
 //
 // [Pulumi Package Schema]: https://www.pulumi.com/docs/guides/pulumi-packages/schema/
-func MainWithMuxer(provider string, infos ...tfbridge.Muxed) {
-	ctx := context.Background()
-
-	contract.Assertf(len(infos) == 2,
-		"We expect to see a SDKv2 and then a provider info for a muxed provider")
-	sdkV2 := infos[0].SDK
-	pf := *infos[1].PF
-
-	shim := AugmentShimWithPF(ctx, sdkV2.P, pf.NewProvider()).(*mergedProviderShim)
+func MainWithMuxer(info sdkBridge.ProviderInfo) {
```

Comment:
I'm guessing this can't merge into Main yet? 

You could rename Main -> classicMain, MainWithMuxer -> mainWithMuxer and write Main that detects `info.P.(*pfmuxer.ProviderShim)` and dispatches based on that.

Reply:
It could, and this would remove our dependence on `pf/tfbridge.ProviderInfo`. My preference would be to add the new functionality under `MainWithMuxer`, making sure that it is well tested and existing tests pass as is.

I'll merge `MainWithMuxer` into `Main` in a follow up PR that necessarily touches all our existing integration tests.

Reply:
Thank you makes sense! We can defer that until "API cleanup" phase later on too - it does not matter for functionality per se.